exp/controllers/awsmachinepool_controller.go

```diff
@@ -393,7 +392,12 @@ func (r *AWSMachinePoolReconciler) reconcileDelete(machinePoolScope *scope.Machi
 
 func (r *AWSMachinePoolReconciler) updatePool(machinePoolScope *scope.MachinePoolScope, clusterScope cloud.ClusterScoper, existingASG *expinfrav1.AutoScalingGroup) error {
 	asgSvc := r.getASGService(clusterScope)
-	if asgNeedsUpdates(machinePoolScope, existingASG) {
+
+	subnetIDs, err := asgSvc.SubnetIDs(machinePoolScope)
+	if err != nil {
+		return errors.Wrapf(err, "fail to get subnets for ASG")
+	}
+	if asgNeedsUpdates(machinePoolScope, existingASG) || !cmp.Equal(subnetIDs, existingASG.Subnets) {
```

Comment:
Hi @Ankitasw , if there are no subnetIDs from machinePoolScope, do you think we should continue the compare and then update if not consistent?